tests/CalculatorTest.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace MF\Bulletproof;
+
+use MF\Bulletproof\Fixture\Calculator;
+
+class CalculatorTest extends AbstractTestCase
+{
+    use \Eris\TestTrait;
+
+    /** @var Calculator */
+    private $calculator;
+
+    protected function setUp(): void
+    {
+        $this->calculator = new Calculator();
+
+        // this should be done by PHPStan
+        $this->init(
+            new class() implements TypeInferenceInterface {
+                public function deduceType(string $object, string $method): array
```

Comment:
would be interesting to see how detailed would the detected types be, e.g. is an array of integers vs. of MyClass instances treated differently?

Reply:
That was indeed part of the discussion: how do we, for example, express that a type is a range of `[0..10]`, or `Natural`, or `Rational`, or `FutureDate`?

Reply:
Well I guess it could be _emulated_ by when method of Eris, at least at the begining.. 

So the type itself would be a nice start ;)